zerver/lib/markdown/__init__.py

```diff
@@ -2438,14 +2449,36 @@ def topic_links(linkifiers_key: int, topic_name: str) -> List[Dict[str, str]]:
                 url = result._replace(scheme="https").geturl()
             else:
                 url = actual_match_url
-            matches.append(dict(url=url, text=actual_match_url, index=pos))
+            matches.append(
+                TopicLinkMatch(
+                    url=url,
+                    text=actual_match_url,
+                    index=pos,
+                    precedence=0,
```

Comment:
Why this value for `precedence`? I think the previous behavior was we had all linkifier matches appear before all raw URLs; I'm not sure what product behavior we want, but this looks like URLs have the same precedence as whatever the first linkifier was?

Reply:
Previously they are mixed up, and ordered by the starting index of the match for both raw URLs and linkifiers. I think it kind of make sense to prioritize raw URLs. Making the precedence -1 might make more sense here. For message body, linkifiers (45 ~ 54) have a lower priority than links (55).

Reply:
This is updated by setting precedence for raw URLs to `null` or `None`, and using the starting index (the order the matches appear) as the tie-breaker when dealing with them.